main.c

```diff
@@ -2,20 +2,24 @@
 
 int main()
 {
+    int print_flags = 0;
     struct slist head;
     struct slist *headp = &head;
+
+    print_flags = PRINT_SLIST_FLAG_PRINT_VAL_ONLY;
```

Comment:
"print_flags" is the argument of a "logger" function. This should not be exposed to the users.
Please provide, "get()", "set()", methods on "print_flags". "print_flags" should be a static int in a file wherever "print" function/macro is defined.